src/apps/allbridge/positions.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  PositionsHook,
+  AppTokenPositionDefinition,
+  UnknownAppTokenError,
+  TokenDefinition,
+  ContractPositionDefinition,
+} from '../../types/positions'
+import { Address } from 'viem'
+import { DecimalNumber, toDecimalNumber } from '../../types/numbers'
+import BigNumber from 'bignumber.js'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { getAllbridgeTokenInfo } from './api'
+import { poolAbi } from './abis/pool'
+
+const ALLBRIDGE_LOGO =
+  'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/allbridgecore.png'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'allbridge',
+      name: 'Allbridge',
+      description: '',
+    }
+  },
+  async getPositionDefinitions(networkId, address) {
+    const allbridgeTokenInfo = (await getAllbridgeTokenInfo({ networkId }))
+      .tokens
+
+    const client = getClient(networkId)
+
+    const balances = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'balanceOf',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
+
+    const rewards = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'pendingReward',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
```

Comment:
this wait we benefit from [client multicall batching](https://valora-app.slack.com/archives/C02898GN22V/p1722510184040949?thread_ts=1722509401.370569&cid=C02898GN22V) for both:
```suggestion
    const [balances, rewards] = await Promise.all([
      await Promise.all(
        allbridgeTokenInfo.map(async (tokenInfo) => {
          return address
            ? client.readContract({
                address: tokenInfo.poolAddress,
                abi: poolAbi,
                functionName: 'balanceOf',
                args: [address as Address],
              })
            : undefined
        }),
      ),
      await Promise.all(
        allbridgeTokenInfo.map(async (tokenInfo) => {
          return address
            ? client.readContract({
                address: tokenInfo.poolAddress,
                abi: poolAbi,
                functionName: 'pendingReward',
                args: [address as Address],
              })
            : undefined
        }),
      )
    ])
```